Merge in (and fix) KeyPhact's U6 implementation

dmadisetti opened the pull request:
We want to do this right?

tschroeder-zendesk committed:
Allow using a proxy.

tejado committed:
Rewrite of google/ptc auth in regard to refresh/access token and addes
Signature.proto

wchill committed:
Update proto

keyphact committed:
Merge pull request #1 from keyphact/fix-proto

Update proto

globeriz committed:
Update Signature.proto

Put semicolons back so it compiles

globeriz committed:
Update Signature_pb2.py

Based on newest proto

globeriz committed:
Update Signature_pb2.py

Fixing path issues

cheesynoob committed:
Update utilities.py

Added functions to generate the hashes and hashArray required for u6

globeriz committed:
iOSActivity should be Activity

cheesynoob committed:
Merge pull request #3 from keyphact/cheesynoob-patch-1

Update utilities.py

globeriz committed:
Update Signature.proto

another line to change

globeriz committed:
Update Signature_pb2.py

update to reflect renaming in proto

cheesynoob committed:
Fixed mistake in generateRequests

forgot to pass in the authticket

KevinCollmer committed:
Merge pull request #4 from keyphact/cheesynoob-patch-2

Fixed mistake in generateRequests

classyjakey committed:
Merge pull request #2 from keyphact/fix-proto-1

Fix Signature.proto

TalSk committed:
Quick fix to utilities functions

Quick fix to utilities functions

JosiahWhite committed:
Merge pull request #5 from keyphact/tal-patch1

Quick fix to utilities functions

tejado committed:
Added wrapper for signature encryption

keyphact committed:
Merge pull request #6 from tejado/master

Added wrapper for signature encryption

cheesynoob committed:
changed request altitude to int value

oops

keyphact committed:
Merge pull request #8 from keyphact/cheesynoob-patch-1

changed request altitude to int value

dmadisetti committed:
d2h fix and options for pokecli

dmadisetti committed:
Merge res

dmadisetti committed:
It works now

dmadisetti committed:
Works now

Alternatively rebase with https://github.com/tejado/pgoapi if he accepts my commit.

wchill committed:
Merge pull request #9 from dmadisetti/patch-1

Works now

pbdeuchler committed:
Use six for chr() call for python3 compat, pep8 fixes

dmadisetti commented on pgoapi/pgoapi.py:
```diff
@@ -106,6 +108,12 @@ def create_request(self):
+    def activate_signature(self, lib_path):    
+        self._signature_lib = lib_path
+
+    def get_signature_lib(self):    
```
Totally preference, but properties are awesome http://docs.quantifiedcode.com/python-anti-patterns/correctness/implementing_java-style_getters_and_setters.html

wchill committed:
Merge pull request #11 from catchemallio/python3

Python3 compatibility and Pep8 fixes

Noctem committed:
Fix hex decoding on Python 3

Use binascii's unhexlify (already imported) to decode the hex since
using `decode` results in an AttributeError in Python 3.

wchill committed:
Merge pull request #23 from Noctem/hexdecode

Fix hex decoding in Python 3

Noctem commented:
For Python 3 compatibility, cherry-pick or merge keyphact/pgoapi@75eba6b5b630841ee4f7c2ea983f15874fb0862d and then Noctem/pgoapi@6b50289299f7517efb1d2ed184cc2dd774c3e60e.

dmadisetti committed:
Merge in Keys and Conflict res

scottstamp committed:
Added scripts\accept-tos.py (#30)

* added scripts\accept-tos.py

* tidied up accept-tos.py

rbignon committed:
correctly return the new access_token after refresh

edevil committed:
Add a sane API timeout.

wchill committed:
Fix index error on API endpoint (#63)

keyphact committed:
Merge pull request #54 from edevil/api_timeout

Add a sane API timeout.

keyphact committed:
Merge pull request #49 from rbignon/master

correctly return the new access_token after refresh

Noctem committed:
Merge proxy support

globeriz committed:
Merge pull request #66 from Noctem/proxies

Add proxy support

OurFlagIsMined committed:
Add auth service as an optional parameter

pengstrom committed:
Added future as a requirement

globeriz committed:
Merge pull request #69 from OurFlagIsMined/patch-1

Add auth service as an optional parameter

globeriz committed:
Merge pull request #70 from pengstrom/pengstrom/add-future-requirement

Added future as a requirement

cheesynoob committed:
Now reusing sessions (#73)

cheesynoob committed:
Fixed missing import (#74)

cheesynoob committed:
Uses auth_info as seed when auth_ticket is unavailable (#74)

* Uses auth_info as seed when auth_ticket is unavailable

This is what the client does too.

* combined if statements

Noctem committed:
Fix proxies with reused sessions (#77)

The proxy code got left behind when sessions were moved to the PGoApi
object, so I applied it to the new sessions and simplified the code.

peter-bonanni commented:
Is this going to get merged?

dmadisetti committed:
Merge branch 'master' of https://github.com/keyphact/pgoapi
